sklearn/model_selection/_split.py

```diff
@@ -1878,6 +1879,87 @@ def get_n_splits(self, X=None, y=None, groups=None):
         return len(self.unique_folds)
 
 
+class BinnedStratifiedKFold(StratifiedKFold):
+    """Stratified K-Folds cross-validator
+
+    Provides train/test indices to split data in train/test sets.
+
+    This cross-validation object is a variation of KFold that returns
+    stratified folds by binning a regression target. The folds are made
+    by approximately preserving the distribution of targets in each bin.
+
+    Read more in the :ref:`User Guide <cross_validation>`.
+
+    Parameters
+    ----------
+    n_splits : int, default=5
+        Number of folds. Must be at least 2.
+
+    shuffle : boolean, default=False
+        Whether to shuffle each stratification of the data before splitting
+        into batches.
+
+    n_bins : int, default=10
```

Comment:
5 to keep consistent with calibration_curve and KBinsDiscretizer?

Reply:
I feel like there's not really a reason to keep it consistent and 10 will be "better" in some sense and I don't see a downside of using 10.

Reply:
> 10 will be "better" in some sense

could you please provide more details?

Reply:
Well it discretizes less and uses more bins?

Reply:
> Well it discretizes less and uses more bins?

In this way, 10 is better than 5 and 20 is better than 10? I guess it depends on the dataset.
Let's set 10 as the default if you like, thoguh I still prefer 5 :)